homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,115 @@
+    """ Gets a list of the available stations and sets up the sensors. """
+
+    client_id = config.get('client_id', None)
+    client_secret = config.get('client_secret', None)
+    username = config.get('username', None)
+    password = config.get('password', None)
+    dev = list()
+    request = requests.post("https://api.netatmo.net/oauth2/token",
+                            data={'grant_type': 'password',
+                                  'client_id': client_id,
+                                  'client_secret': client_secret,
+                                  'username': username,
+                                  'password': password})
+    access_token = request.json()["access_token"]
+    request = requests.get(sensor_url + access_token).json()
+    for device in request["body"]["devices"]:
+        for data_type in device["data_type"]:
+            dev.append(NetatmoSensor(hass, client_id, client_secret,
+                                     username, password, data_type,
+                                     device["module_name"]))
+        for module in device["modules"]:
+            for data_type in module["data_type"]:
+                dev.append(NetatmoSensor(hass, client_id, client_secret,
+                                         username, password, data_type,
+                                         module["module_name"]))
+    add_devices(dev)
+
+
+# pylint: disable=too-many-arguments
+class NetatmoSensor(Entity):
+    """ Implements a netatmo sensor. """
+
+    def __init__(self, hass, client_id, client_secret, username,
+                 password, data_type, module_name):
+        self._hass = hass
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._username = username
+        self._password = password
+        self._state = STATE_UNKNOWN
+        self._data_type = data_type
+        self._name = module_name
+        self._unit_of_measurement = SENSOR_TYPES[data_type][0]
+        self.update()
+
+    @property
+    def name(self):
+        """ The name of the sensor. """
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit the value is expressed in. """
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """ Gets the latest data from Netatmo API and updates the state. """
+        request = requests.post("https://api.netatmo.net/oauth2/token",
+                                data={'grant_type': 'password',
+                                      'client_id': self._client_id,
+                                      'client_secret': self._client_secret,
+                                      'username': self._username,
+                                      'password': self._password})
+        access_token = request.json()["access_token"]
```

Comment:
Perhaps it would be possible to re-use the token from the setup? Depending on how long the token is valid, a request every 5 min could be saved.